pandas/io/packers.py

```diff
@@ -402,13 +403,17 @@ def encode(obj):
                     u'freq': u_safe(getattr(obj, 'freqstr', None)),
                     u'tz': tz,
                     u'compress': compressor}
-        elif isinstance(obj, IntervalIndex):
-            return {u'typ': u'interval_index',
+        elif isinstance(obj, (IntervalIndex, IntervalArray)):
```

Comment:
Should talk about this more generally in https://github.com/pandas-dev/pandas/issues/20612

Reply:
Mmm, that doesn't sound fun. I'll see what I can do to avoid that.

Reply:
I wonder, should this be reverted? Do any msgpack tests fail if it is?

Reply:
The test below fails on the last copied line for `Interval` data if reverted:

https://github.com/pandas-dev/pandas/blob/da6e26db079cd4050c1693939db6a52afb2978e1/pandas/tests/io/test_packers.py#L357-L360